### PR TITLE
Feature/dact 775 check your answers director details

### PR DESCRIPTION
**Pull request by jbestwickch**
https://companieshouse.atlassian.net/browse/DACT-775
Add directors details content for appoint check your answers.
Logic added to ensure continue and back-links redirect to check your answers page if a page is reached from the change link on check your answers.
Requires the changes in https://github.com/companieshouse/officer-filing-api/pull/146

### CODE REVIEW

File: src/utils/web.ts [hunk 1]
```diff
@@ -10,6 +15,36 @@ export const getField = (req: Request, fieldName: string): string => {
...
+  }
+  return backLink;
+};
+
+/**
+ * Set the redirect link for a page. 
+ * Checks whether the user came from the check your answers page and if so,
+ *  sets the redirect link to the check your answers page instead.
+ */
+export const setRedirectLink = async (req: Request, checkYourAnswersLink: string | undefined, backLink: string): Promise<string> => {
```
Comment (mdesaikns) [inline src/utils/web.ts:35]: should this variable name backLink be redirectLink?
just pointing it out as I had a hard time understanding it in controller classes.

File: test/mocks/check.your.answers.middleware.mock.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { NextFunction, Request, Response } from "express";
+import { checkYourAnswersMiddleware } from "../../src/middleware/check.your.answers.middleware";
+
+// get handle on mocked function
+const mockcheckYourAnswersMiddleware = checkYourAnswersMiddleware as jest.Mock;
+
+// tell the mock what to return
+mockcheckYourAnswersMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next());
+
+export default mockcheckYourAnswersMiddleware;
```
Comment (mdesaikns) [inline test/mocks/check.your.answers.middleware.mock.ts:12]: is this used?

File: src/utils/web.ts (hunk 1 above)
Comment (jbestwickch) [reply src/utils/web.ts:35]: Yeah good point, must have copied the method and not updated.

File: test/mocks/check.your.answers.middleware.mock.ts (hunk 2 above)
Comment (jbestwickch) [reply test/mocks/check.your.answers.middleware.mock.ts:12]: No was not needed in the end. Will remove

**Review by mdesaikns — APPROVED** [review]

**Review by clewis1 — APPROVED** [review]
Looks good.

**Pull request merged**